_posts/2020-12-17-Resilience4j-grpc.md

```diff
@@ -0,0 +1,187 @@
+---
+layout: post
+title: Setting up Resilience4j Circuit Breaker for Grpc Java Client
+author: alicanhaman
+categories: [grpc, resilience4j]
+tags: [grpc, resilience4j, spring boot]
+---
+
+In this article, I will explain how to set up [Resilience4j](https://github.com/resilience4j/resilience4j) circuit breaker to work with a grpc client set up with [grpc-spring-boot-starter](https://github.com/yidongnan/grpc-spring-boot-starter). Normally, resilience4j is built to handle restful requests easily, but when it comes to grpc is [falls short](https://github.com/resilience4j/resilience4j/issues/1067). There is an open PR in the resilience4j repository, but it seems to have been left to rot.
+
+First lets briefly mention our stack: 
+
+- **Resilience4j** is a lightweight fault tolerance library inspired by [Netflix Hystrix](https://github.com/Netflix/Hystrix), but designed for Java 8 and functional programming. Lightweight, because the library only uses Vavr, which does not have any other external library dependencies. Netflix Hystrix, in contrast, has a compile dependency to Archaius which has many more external library dependencies such as Guava and Apache Commons Configuration. Since Hytrix have been moved to about 2 years ago, this library is deemed as a good alternative.
+- **gRPC** is a modern open source high performance RPC framework that can run in any environment. It can efficiently connect services in and across data centers with pluggable support for load balancing, tracing, health checking and authentication.
+
+## Dependencies
+
+I used maven for dependency management, here is how a simple pom would like.
+
+``` xml
+<dependencies>
+  <dependency>
+      <groupId>net.devh</groupId>
+      <artifactId>grpc-client-spring-boot-starter</artifactId>
+  </dependency>
+  <!-- basically your grpc proto project -->
```

Comment:
I'd give more detail regarding proto and grpc relation

Reply:
Wrote some content on that and gave reference to baeldung guide.